extension/ackextension/inmemory_test.go

```diff
@@ -0,0 +1,247 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ackextension
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAckPartitionNextAckConcurrency(t *testing.T) {
+	ackSize := 1_000_000
+	ap := newAckPartition(uint64(ackSize))
+	map1 := map[uint64]struct{}{}
+	map2 := map[uint64]struct{}{}
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		for i := 0; i < ackSize/2; i++ {
+			map1[ap.nextAck()] = struct{}{}
+		}
+		wg.Done()
+	}()
+	go func() {
+		for i := 0; i < ackSize/2; i++ {
+			map2[ap.nextAck()] = struct{}{}
+		}
+		wg.Done()
+	}()
+
+	wg.Wait()
+
+	for k, v := range map2 {
+		map1[k] = v
+	}
```

Comment:
Since go 1.21 we can use `maps.Copy(map1, map2)`

Reply:
changed